Dispenser stuff

supersaiyansubtlety opened the pull request:
- completes `n/m/block/dispenser/` and `CauldronBehavior`
- renames `ItemDispenserBehavior` -> `DefaultDispenserBehavior`
- rename `FallibleItemDispenserBehavior` -> `FallibleDispenserBehavior`
- renames  `n/m/util/math/BlockPointer` -> `n/m/block/dispenser/DispenserContext`:
it holds a `DispenserBlockEntity` and is used by `DispenserBehavior`s, context is more conventional than pointer
- consistently names the methods called in `Bootstrap::bootstrap` `bootstrap`

supersaiyansubtlety committed:
rename n/m/util/math/BlockPointer -> n/m/block/dispenser/DispenserContext:
it holds a DispenserBlockEntity and is used by DispenserBehaviors, context is more conventional than pointer

supersaiyansubtlety committed:
complete and clarify BlockHitResult

supersaiyansubtlety committed:
map MinecartDispenserBehavior

supersaiyansubtlety committed:
add simple type for Hand

supersaiyansubtlety committed:
map EquipmentDispenserBehavior

supersaiyansubtlety committed:
complete n/m/block/dispenser
complete CauldronBehavior
rename Bootstrap ->  Bootstrapper and make bootstrap methods names consistent

supersaiyansubtlety committed:
rename    ItemDispenserBehavior -> DefaultDispenserBehavior
rename    FallibleItemDispenserBehavior -> FallibleDispenserBehavior

eristhea commented on mappings/net/minecraft/Bootstrapper.mapping:
```diff
@@ -1,5 +1,5 @@
-CLASS net/minecraft/unmapped/C_ntskruzf net/minecraft/Bootstrap
-	FIELD f_jltdqkzo initialized Z
+CLASS net/minecraft/unmapped/C_ntskruzf net/minecraft/Bootstrapper
```
i disagree with this rename, since it's an all-static class. additionally, `Bootstrap` has parity across yarn and moj

supersaiyansubtlety replied:
I mainly did it to avoid `Bootstrap.bootstrap()`, bit ig that's fine.

Do you prefer the old "initialize" field+method names as well?

eristhea replied:
i think the new ones are better!

eristhea reviewed (changes requested):
looks great, just one issue!

supersaiyansubtlety committed:
restore old name Bootstrapper -> Bootstrap

supersaiyansubtlety commented on src/packageDocs/java/net/minecraft/package-info.java:
```diff
@@ -7,7 +7,7 @@
  *
  * <p>Unmapped classes go into this package by default. This package additionally contains
- * {@link Bootstrap}, {@link SharedConstants}, and {@link MinecraftVersion} classes.
+ * {@link com.sun.jdi.Bootstrap}, {@link SharedConstants}, and {@link MinecraftVersion} classes.
```
oops, fixing

supersaiyansubtlety committed:
Merge branch '1.21.4-pre2' into dispenser-stuff

github-actions[bot] committed:
Merge branch '1.21.4-rc3' into dispenser-stuff

actions-user committed:
Drop invalid mappings

supersaiyansubtlety committed:
re-complete DispenserBehavior
rename DefaultDispenserBehavior dispenseSilently -> dispenseImpl and correct its old pointer param name

github-actions[bot] committed:
Merge branch '1.21.4' into dispenser-stuff

actions-user committed:
Drop invalid mappings

github-actions[bot] committed:
Merge branch '25w02a' into dispenser-stuff

IotaBread commented on mappings/net/minecraft/world/WorldEvents.mapping:
Javadocs in this class still use `n.m.util.math.BlockPointer` instead of `n.m.block.dispenser.DispenserContext`

supersaiyansubtlety replied:
fixed

supersaiyansubtlety committed:
update javadoc references in WorldEvents
